src/pocket/PocketApiGatewayLambdaIntegration.ts

```diff
@@ -0,0 +1,106 @@
+import { APIGateway, LambdaFunction } from '@cdktf/provider-aws';
+import { Resource } from '@cdktf/provider-null';
+import { Construct } from 'constructs';
+
+import { PocketVersionedLambda, PocketVersionedLambdaProps } from '..';
+
+export interface ApiGatewayLambdaRoute {
+  path: string;
+  // The HTTP method clients use to call this method (GET, POST, etc.)
+  method: string;
+  // Currently not supporting authorizer
+  // authorizationType: string;
+  eventHandler: PocketVersionedLambdaProps;
+  // TODO
+  // requestParams: Map<string, any>;
+}
+
+export interface PocketApiGatewayProps {
+  name: string;
+  routes: ApiGatewayLambdaRoute[];
+}
+
+interface InitializedGatewayRoute {
+  resource: APIGateway.ApiGatewayResource;
+  method: APIGateway.ApiGatewayMethod;
+  integration: APIGateway.ApiGatewayIntegration;
+  permission: LambdaFunction.LambdaPermission;
+}
+
+export class PocketApiGateway extends Resource {
+  private readonly config: PocketApiGatewayProps;
+  private apiGatewayRestApi: APIGateway.ApiGatewayRestApi;
+  private routes: InitializedGatewayRoute[];
+  private apiGatewayDeployment: APIGateway.ApiGatewayDeployment;
+  private apiGatewayStage: APIGateway.ApiGatewayStage;
```

Comment:
If I am remembering my api-gateways facts correctly I think you need to do something with the stage to map that to a deployment? I think as is you don't have a `stage` which means your resources are deployed but no stage mapping that to a deployment meaning you cannot access the `test-api-lambda` endpoint externally

Reply:
gotcha, I can add the deployment 👍

Reply:
I added the deployment and stage resources, but I'm not sure that the triggers are managed correctly. See code comments... I followed the example in the docs [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment#terraform-resources). I tried to remove the `id` attribute to compute the hash of whole resources, but that led to a StackOverflow error.

Reply:
I like your idea! I remember this being very odd last time I implemented an api-gateway in terraform. I remember doing something like [this](https://github.com/hashicorp/terraform-provider-aws/issues/162#issuecomment-532593939) to solve it before. An option here might be a sha1 of the config object passed in? I think that would cover all changes to the configuration?
```typescript
Fn.sha1(Fn.jsonencode(config))
```
```

Reply:
ooh, the config. That's a good idea.